runtime/jcl/common/jclcinit.c

```diff
@@ -305,7 +305,7 @@ jint initializeKnownClasses(J9JavaVM* vm, U_32 runtimeFlags)
 					return JNI_ERR;
 				}
 			}
-		} else if (J9CPTYPE_INSTANCE_METHOD == J9_CP_TYPE(cpShapeDescription, i)) {
+		} else if (J9CPTYPE_INSTANCE_METHOD == J9_CP_TYPE(cpShapeDescription, i) || J9CPTYPE_INTERFACE_INSTANCE_METHOD == J9_CP_TYPE(cpShapeDescription, i)) {
```

Comment:
Both `J9CPTYPE_INTERFACE_INSTANCE_METHOD` & `J9CPTYPE_INTERFACE_STATIC_METHOD` should never happen for the VM's constantpool.  We should have both of these treated as an error if they ever occurred.

For background, the VM generates a constantpool of data - references to class and {static, instance}x{fields, methods} and should never generate this kind of incorrect data.

Reply:
`J9CPTYPE_INTERFACE_INSTANCE_METHOD & J9CPTYPE_INTERFACE_STATIC_METHOD` are a subset of `J9CPTYPE_INSTANCE_METHOD & J9CPTYPE_STATIC_METHOD` where it have a classfile tag of `CFR_CONSTANT_InterfaceMethodref`, not sure why they are incorrect.

FYI, the new types are added to carry extra data (since we remap the classfile methodrefs) for resolve checks. They are not meant as incorrect.

Reply:
Sorry, I was wrong about the purpose of this change.  Leaving it as is should be fine.

Reply:
This has the same formatting concern as above.  Missing `()` on the individual clauses and would be better as 3 lines.